### ISSUE TITLE
`ExId` is poor naming

### ISSUE THREAD

**Issue by teohhanhui** [automerge/automerge#754]
It makes sense internally for the library, but an "external ID" is redundant naming for consumers of this library.

**Comment by alexjg** [comment]
It's actually exported as `Objid` but various parts of the rust ecosystem are not good at using the exported name (e.g. see https://docs.rs/automerge/latest/automerge/enum.ObjId.html)

**Comment by teohhanhui** [comment]
Even that docs.rs page uses `ExId` throughout 🙈

Would it make sense for everything to reference `crate::ObjId` instead? (Not sure if that'd help.)

**Comment by alexjg** [comment]
Yeah `docs.rs` is an example of the exported name not being respected for some reason.

`crate::ObjId` is crate private because it's not safe to use between docs (the actor index it uses internally is specific to the document it came from) so we definitely need the `ExID` type. It might work to rename the type to `ObjId` and everywhere we use it internally to `use crate::exid::ObjId as ExId`.

**Comment by teohhanhui** [comment]
I think #755 should be fine?

**Comment by FintanH** [comment]
Closing this, referencing this comment https://github.com/automerge/automerge/pull/755#issuecomment-1747326956.
